webapp/components/post_view/components/post_attachment_opengraph.jsx

```diff
@@ -0,0 +1,192 @@
+import * as Utils from 'utils/utils.jsx';
+import * as CommonUtils from 'utils/commons.jsx';
+import {updateOpenGraphMetaData} from 'actions/global_actions.jsx';
+
+export default class PostAttachmentOpenGraph extends React.Component {
+    constructor(props) {
+        super(props);
+        this.imageDimentions = {  // Image dimentions in pixels.
+            height: 150,
+            width: 150
+        };
+        this.maxDescriptionLength = 300;
+        this.descriptionEllipsis = '...';
+        this.fetchData = this.fetchData.bind(this);
+        this.onOpenGraphDataChange = this.onOpenGraphDataChange.bind(this);
+        this.toggleImageVisibility = this.toggleImageVisibility.bind(this);
+        this.onImageLoad = this.onImageLoad.bind(this);
+    }
+
+    componentWillMount() {
+        this.setState({
+            data: {},
+            imageLoaded: false,
+            imageVisible: this.props.previewCollapsed.startsWith('false')
+        });
+        this.fetchData(this.props.link);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setState({imageVisible: nextProps.previewCollapsed.startsWith('false')});
+        if (!Utils.areObjectsEqual(nextProps.link, this.props.link)) {
+            this.fetchData(nextProps.link);
+        }
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        if (nextState.imageVisible !== this.state.imageVisible) {
+            return true;
+        }
+        if (nextState.imageLoaded !== this.state.imageLoaded) {
+            return true;
+        }
+        if (!Utils.areObjectsEqual(nextState.data, this.state.data)) {
+            return true;
+        }
+        return false;
+    }
+
+    componentDidMount() {
+        OpenGraphStore.addUrlDataChangeListener(this.onOpenGraphDataChange);
+    }
+
+    componentDidUpdate() {
+        if (this.props.childComponentDidUpdateFunction) {
+            this.props.childComponentDidUpdateFunction();
+        }
+    }
+
+    componentWillUnmount() {
+        OpenGraphStore.removeUrlDataChangeListener(this.onOpenGraphDataChange);
+    }
+
+    onOpenGraphDataChange(url) {
+        if (url === this.props.link) {
+            this.fetchData(url);
+        }
+    }
+
+    fetchData(url) {
+        const data = OpenGraphStore.getOgInfo(url);
+        this.setState({data, imageLoaded: false});
+        if ($.isEmptyObject(data)) {
+            updateOpenGraphMetaData(url);
+        }
+    }
+
+    getBestImageUrl() {
+        const nearestPointData = CommonUtils.getNearestPoint(this.imageDimentions, this.state.data.images, 'width', 'height');
+
+        const bestImage = nearestPointData.nearestPoint;
+        const bestImageLte = nearestPointData.nearestPointLte;  // Best image <= 150px height and width
+
+        const finalBestImage = (
+            !$.isEmptyObject(bestImageLte) && bestImageLte.height <= this.imageDimentions.height && bestImageLte.width <= this.imageDimentions.width ?
```

Comment:
We're trying to move away from using jQuery for a few reasons. Would you be able to add a utils function that checks for an empty object without using jQuery and use that here instead?

I think EMCA5 method `Object.keys(obj).length === 0 && obj.constructor === Object` should work